Unable to create a socks5 hyper client

Dear hyper-socks2 community,

I am trying to use this crate to create a hyper client with socks5 connection as below

```
use hyper::Uri;
use hyper_socks2::SocksConnector;
use hyper_util::client::legacy::connect::HttpConnector;

...

        let mut http_connector: HttpConnector = HttpConnector::new();
        http_connector.enforce_http(false);

        let socks_http_connector = SocksConnector {
            proxy_addr: std::env::var("SOCKS5").unwrap().parse::<Uri>().unwrap(), // scheme is required by HttpConnector
            auth: None,
            connector: http_connector.clone(),

        let http_client = HyperClientBuilder::new().build(socks_http_connector);
```

This code fails with compilation error:

```
error[E0277]: the trait bound `SocksConnector<HttpConnector>: tower::Service<http::uri::Uri>` is not satisfied
   --> src/ims/s3.rs:99:59
    |
99  |         let http_client = HyperClientBuilder::new().build(socks_http_connector);
    |                                                     ----- ^^^^^^^^^^^^^^^^^^^^ the trait `tower::Service<http::uri::Uri>` is not implemented for `SocksConnector<HttpConnector>`
    |                                                     |
    |                                                     required by a bound introduced by this call
    |
    = help: the trait `tower::Service<Uri>` is implemented for `SocksConnector<HttpConnector>`
    = help: for that trait implementation, expected `Uri`, found `http::uri::Uri`
note: required by a bound in `HyperClientBuilder::build`
   --> /home/msopena/.cargo/registry/src/index.crates.io-6f17d22bba15001f/aws-smithy-runtime-1.7.3/src/client/http/hyper_014.rs:597:12
    |
594 |     pub fn build<C>(self, tcp_connector: C) -> SharedHttpClient
    |            ----- required by a bound in this associated function
...
597 |         C: hyper_0_14::service::Service<http_02x::Uri>,
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `HyperClientBuilder::build`
```

And my dependencies:

```
hyper-socks2 = "0.9.1" 
hyper = "1.5.0"
hyper-util = "0.1.9"
hyper-rustls = { version = "0.27.3", features = ["webpki-roots"] }
```

I would like to know how to create a socks5 client using this crate

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -4,9 +4,14 @@
 //! ```no_run
 //! # use std::error::Error;
 //! # fn hidden() -> Result<(), Box<dyn Error>> {
-//! use hyper::{Body, Uri};
-//! use hyper::client::{Client, HttpConnector};
+//! use bytes::Bytes;
+//! use http_body_util::Full;
+//! use hyper::Uri;
 //! use hyper_socks2::SocksConnector;
+//! use hyper_util::{
+//!     client::legacy::{connect::HttpConnector, Client},
+//!     rt::TokioExecutor,
+//! };
 //!
 //! let mut connector = HttpConnector::new();
 //! connector.enforce_http(false);
@@ -19,7 +24,7 @@
 //! // with TLS support
 //! let proxy = proxy.with_tls()?;
 //!
-//! let client = Client::builder().build::<_, Body>(proxy);
+//! let client = Client::builder(TokioExecutor::new()).build::<_, Full<Bytes>>(proxy);
 //!
 //! # Ok(())
 //! # }

